python/dask_cudf/dask_cudf/core.py

```diff
@@ -90,11 +91,30 @@ def __init__(self, dsk, name, meta, divisions):
         self._meta = meta
         self.divisions = tuple(divisions)
 
-    def __getstate__(self):
-        return (self.dask, self._name, self._meta, self.divisions)
-
-    def __setstate__(self, state):
-        self.dask, self._name, self._meta, self.divisions = state
+    def serialize(self):
+        meta_header, meta_frames = self._meta.serialize()
+        header = {
+            "dask": pickle.dumps(self.dask),
+            "name": self._name,
+            "meta": {
+                "type": pickle.dumps(type(self._meta)),
+                "header": meta_header,
+            },
+            "divisions": self.divisions,
+        }
+        frames = meta_frames
+        return header, frames
+
+    @classmethod
+    def deserialize(cls, header, frames):
+        dsk = pickle.loads(header["dask"])
+        name = header["name"]
+        divisions = header["divisions"]
+        meta_type = pickle.loads(header["meta"]["type"])
+        meta_header = header["meta"]["header"]
+        meta_frames = frames
+        self._meta = meta_type.deserialize(meta_header, meta_frames)
```

Comment:
Can't have `self` here.

Reply:
Oops 😬 This is what I get for copy-paste editing my way through this 😂 Thanks for catching that Rick!